### ISSUE TITLE
Instrumented Task instantiated directly now throws TaskInstantiationException

### ISSUE THREAD

**Issue by nbarrett** [serenity-bdd/serenity-core#1433]
Since upgrading to Serenity 2.x, Tasks that are instantiated directly now throw `TaskInstantiationException` with the following message: `Could not instantiate class <Task>. If you are not instrumenting a Task class explicitly you need to give the class a default constructor.`

This now prevents me from running such tasks with/without instrumentation in the reporting as there are some cases where I want to suppress this task from being reported upon. Such an example of this is a Utility waiting Task as follows:

```
public class WaitBuilder {


    private Duration duration = Duration.ofMillis(serenityPropertyAsInteger(WEBDRIVER_TIMEOUTS_IMPLICITLYWAIT));
    private boolean quietly;

    public WaitBuilder forDuration(Duration duration) {
        this.duration = duration;
        return this;
    }

    public WaitBuilder quietly() {
        this.quietly = true;
        return this;
    }

    public <E> Interaction forCondition(ExpectedCondition<E> expectedCondition) {
        return quietly ? new WaitForCondition(duration, expectedCondition) : instrumented(WaitForCondition.class, duration, expectedCondition);
    }
```
This allows me to call the task with instrumentation as follows:

 `actor.attemptsTo(Wait.shortAsyncDuration().forCondition(<expected condition>));`

Or without instrumentation as follows:

 `actor.attemptsTo(Wait.shortAsyncDuration().quietly().forCondition(<expected condition>));`

Is there a way to achieve the above without the `TaskInstantiationException` being thrown?
It makes no sense to have a default constructor either as parameters are mandatory for the task to work.

**Comment by wakaleo** [comment]
I think there was a hard reason for this (can't remember what it was now) - have a look at the code to see if you can find a better way.

**Comment by wakaleo** [comment]
Alternatively the logic could just not instrument classes that don't have a default constructor, though that might be harder to understand.

**Comment by wakaleo** [comment]
True, but those commits had their reasons too, so removing them would break behaviour described in the docs
